qemu/qemu_s390x.go

```diff
@@ -0,0 +1,121 @@
+// +build s390x
+
+/*
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+package qemu
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+// -pci devices don't play well with Z hence corresponding -ccw devices
+// See https://wiki.qemu.org/Documentation/Platforms/S390X
+const (
+	// NVDIMM is the Non Volatile DIMM device driver.
+	NVDIMM DeviceDriver = "nvdimm"
+
+	// Virtio9P is the 9pfs device driver.
+	Virtio9P DeviceDriver = "virtio-9p-ccw"
+
+	// VirtioNet is the virt-io networking device driver.
+	VirtioNet DeviceDriver = "virtio-net"
+
+	// VirtioNetPCI is the virt-io pci networking device driver.
+	VirtioNetPCI DeviceDriver = "virtio-net-ccw"
+
+	// VirtioSerial is the serial device driver.
+	VirtioSerial DeviceDriver = "virtio-serial-ccw"
+
+	// VirtioBlock is the block device driver.
+	VirtioBlock DeviceDriver = "virtio-blk"
+
+	// Console is the console device driver.
+	Console DeviceDriver = "virtconsole"
+
+	// VirtioSerialPort is the serial port device driver.
+	VirtioSerialPort DeviceDriver = "virtserialport"
+
+	// VHostVSockPCI is the vhost vsock pci driver.
+	VHostVSockPCI DeviceDriver = "vhost-vsock-ccw"
```

Comment:
This grates me. But I understand why you did it. Maybe we could transition all the users to a more generic name and eventually get rid of this?

Reply:
Add a new variable VhostVSOCK...where do I include the deprecation notice?